bulkclient/src/main/java/au/csiro/pathling/export/ws/BulkExportRequest.java

```diff
@@ -0,0 +1,278 @@
+ * Request Query Parameters</a>
+ */
+@Value
+@Builder()
+public class BulkExportRequest {
+
+  /**
+   * The level of the export operation.
+   */
+  public interface Level {
+
+    /**
+     * The path to the export operation corresponding to this level.
+     *
+     * @return the path to the export operation.
+     */
+    @Nonnull
+    String getPath();
+
+    /**
+     * Whether this level supports patient-specific exports.
+     *
+     * @return true if patient-specific exports are supported.
+     */
+    boolean isPatientSupported();
+  }
+
+  /**
+   * Represents the system level of the export operation.
+   */
+  @Value
+  public static class SystemLevel implements Level {
+
+    @Nonnull
+    @Override
+    public String getPath() {
+      return "$export";
+    }
+
+    @Override
+    public boolean isPatientSupported() {
+      return false;
+    }
+  }
+
+  /**
+   * Represents the patient level of the export operation.
+   */
+  @Value
+  public static class PatientLevel implements Level {
+
+
+    @Nonnull
+    @Override
+    public String getPath() {
+      return "Patient/$export";
+    }
+
+    @Override
+    public boolean isPatientSupported() {
+      return true;
+    }
+  }
+
+  /**
+   * Represents the group level of the export operation.
+   */
+  @Value
+  public static class GroupLevel implements Level {
+
+    /**
+     * The ID of the group.
+     */
+    @Nonnull
+    String id;
+
+    @Nonnull
+    @Override
+    public String getPath() {
+      return String.format("Group/%s/$export", id);
+    }
+
+    @Override
+    public boolean isPatientSupported() {
+      return true;
+    }
+  }
+
+  /**
+   * The level of the export operation. The default is {@link SystemLevel}.
+   */
+  @Nonnull
+  @Builder.Default
+  Level level = new SystemLevel();
+
+  /**
+   * The format of the output. The value of the '_outputFormat' query parameter.
+   */
+  @Nullable
+  @Builder.Default
+  String _outputFormat = null;
+
+  /**
+   * The date and time to use as the lower bound for the export. The value of the '_since' query
+   */
+  @Nullable
+  @Builder.Default
+  Instant _since = null;
+
+  /**
+   * The types of resources to export. The value of the '_type' query parameter.
+   */
+  @Nonnull
+  @Builder.Default
+  List<String> _type = Collections.emptyList();
+
+
+  /**
+   * The elements to include in the export. The value of the '_elements' query parameter.
+   */
+  @Nonnull
+  @Builder.Default
+  List<String> _elements = Collections.emptyList();
+
+  /**
+   * The criteria to filter the resources to include in the export . The value of the '_typeFilter'
+   * query parameter.
+   */
+  @Nonnull
+  @Builder.Default
+  List<String> _typeFilter = Collections.emptyList();
+
+
+  /**
+   * The patient(s) to include in the export. The value of the 'patient'  parameter.
+   */
+  @Nonnull
+  @Builder.Default
+  List<Reference> patient = Collections.emptyList();
```

Comment:
Is this the full list of parameters that I can use? Is it possible to pass parameters that are not in this class, e.g. `includeAssociatedData`?

Reply:
Yes, this is the full list.
Actually whatever is defined in `BulkExportClient` but this is then mapped to the `BulkExportRequest`.
I think `includeAssociatedData` is the only one missing from the spec.

Reply:
I think we should add this to make the list complete.

Reply:
Implemented.